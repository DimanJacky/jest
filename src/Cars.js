import React from 'react';

const Cars = () => (
    <div className="Cars">
        Cars 111
    </div>
);

export default Cars;
