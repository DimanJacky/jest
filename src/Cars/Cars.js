import React from 'react';
import Car from "./Car/Car";

const Cars = (props) => (
    <ul className="Cars">
        {
            props.cars.map(({name,year},i) => (
                <Car key={i} name={name} year={year} />
            ))
        }
    </ul>
);

export default Cars;
