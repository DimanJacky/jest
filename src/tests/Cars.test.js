import React from 'react';
import Cars from "../Cars/Cars";
import {shallow} from 'enzyme';

describe('Testing Cars', () => {
    it('Cars have rendered correctly', () => {
        const cars = [
            {name: 'audi', year: 2018},
            {name: 'chevrolet', year: 2017},
            {name: 'mazda', year: 2020},
        ];
        const car = shallow(<Cars cars={cars} />);
        expect(car).toMatchSnapshot();
    });
});
