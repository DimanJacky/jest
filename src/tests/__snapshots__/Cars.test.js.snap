exports[`Testing Cars Cars have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cars
    cars={
        Array [
            Object {
              "name": "audi",
              "year": 2018,
            },
            Object {
              "name": "chevrolet",
              "year": 2017,
            },
            Object {
              "name": "mazda",
              "year": 2020,
            },
          ]
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Car
          name="audi"
          year={2018} />,
        <Car
          name="chevrolet"
          year={2017} />,
        <Car
          name="mazda"
          year={2020} />,
      ],
      "className": "Cars",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "name": "audi",
          "year": 2018,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "name": "chevrolet",
          "year": 2017,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "name": "mazda",
          "year": 2020,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Car
            name="audi"
            year={2018} />,
          <Car
            name="chevrolet"
            year={2017} />,
          <Car
            name="mazda"
            year={2020} />,
        ],
        "className": "Cars",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "name": "audi",
            "year": 2018,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "name": "chevrolet",
            "year": 2017,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "name": "mazda",
            "year": 2020,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
